src/wallet/external_signer_scriptpubkeyman.cpp

```diff
@@ -63,3 +63,42 @@ bool ExternalSignerScriptPubKeyMan::DisplayAddress(const CScript scriptPubKey, c
     return false;
 #endif
 }
+
+// If sign is true, transaction must previously have been filled
+TransactionError ExternalSignerScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psbt, int sighash_type, bool sign, bool bip32derivs, int* n_signed) const
+{
+#ifdef ENABLE_EXTERNAL_SIGNER
+    if (!sign) {
+        return DescriptorScriptPubKeyMan::FillPSBT(psbt, sighash_type, false, bip32derivs, n_signed);
+    }
+
+    // Already complete if every input is now signed
+    bool complete = true;
+    for (const auto& input : psbt.inputs) {
+        // TODO: for multisig wallets, we should only care if all _our_ inputs are signed
+        complete &= PSBTInputSigned(input);
+    }
+    if (complete) return TransactionError::OK;
+
+    const std::string command = gArgs.GetArg("-signer", ""); // DEFAULT_EXTERNAL_SIGNER);
+    if (command == "") throw std::runtime_error(std::string(__func__) + ": restart bitcoind with -signer=<cmd>");
+
+    std::string chain = gArgs.GetChainName();
+    const bool mainnet = chain == CBaseChainParams::MAIN;
+    std::vector<ExternalSigner> signers;
+    ExternalSigner::Enumerate(command, signers, mainnet);
+    if (signers.empty()) return TransactionError::EXTERNAL_SIGNER_NOT_FOUND;
+    // TODO: add fingerprint argument in case of multiple signers
+    ExternalSigner signer = signers[0];
```

Comment:
In 57b26f44b9109eb0c366adeb54bf5ebfc816b45e "rpc: send: support external signer"

Perhaps use `GetExternalSigner()` instead of duplicate code?